Lowercase index and table names during hint extraction

jusjosj opened the pull request:
### Description

[Describe what this change achieves - Guidelines below (please delete the guidelines after writing the PR description)]

Without this change there will be a mismatch between the index name used at query time and query creation since index and table are lowercase when the index is being created.

### Issues Resolved

[BABEL-3759] 

Signed-off-by: Justin Jossick [jusjosj@amazon.com](mailto:jusjosj@amazon.com)

### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

jusjosj committed:
[BABEL-3759] Lowercase index and table names during hint extraction

Without this change there will be a mismatch between the index name used
at query time and query creation since index and table are lowercase
when the index is being created.

2jungkook commented on contrib/babelfishpg_tsql/src/tsqlIface.cpp:
```diff
@@ -3535,7 +3535,12 @@ std::string extractIndexValues(std::vector<TSqlParser::Index_valueContext *> ind
+
+			//lowercase index and table names since they are created as lowercase in pg
+			transform(indexName.begin(), indexName.end(), indexName.begin(), ::tolower);
+			transform(table_name.begin(), table_name.end(), table_name.begin(), ::tolower);
```
Please move this out of the for-loop.

2jungkook commented on test/JDBC/expected/BABEL-3759.out:
```diff
@@ -0,0 +1,66 @@
+~~START~~
+text
+Query Text: SELECT/*+ IndexScan(BABEL_3759_t1 babel_3759_idx_t1_varcharbabel_6dae4237a7aa37530f3901c640195669) */ c1_INT FROM BABEL_3759_t1                                    WHERE c2_VARCHAR = 'S'
+Bitmap Heap Scan on babel_3759_t1  (cost=4.18..12.64 rows=4 width=4)
```
Let's remove cost from output.

2jungkook commented on test/JDBC/expected/BABEL-3759.out:
```diff
@@ -0,0 +1,66 @@
+EXEC sp_babelfish_configure 'enable_pg_hint', 'on'
```
Please change the filename to `index-hint-vu-XXX` make this as a form of MVU test.

2jungkook commented on test/JDBC/expected/BABEL-3759.out:
```diff
@@ -0,0 +1,66 @@
+SET babelfish_showplan_all ON
+GO
+
+SELECT c1_INT FROM BABEL_3759_t1 (INDEX(BABEL_3759_idx_t1_VARCHAR)) WHERE c2_VARCHAR = 'S'
```
Every hint tests should contain two results (with hint, without hint) to verify it more clearly.

jusjosj committed:
[BABEL-3759] Lowercase hints since tables in pg_hint are case sensitive

In addition to lowercasing the table and index name before hashing, we
also need to lower case the table name in the hints themselves.  Without
this change hints would not be applied when tables with uppercase
letters are in present in the FROM clause.  Lowercase the whole hint has
ensures that tables are always lowercase.

jusjosj committed:
[BABEL-3759] Remove unneeded test file

Tests are covered in existing tests
